server/user/bitrot-detect

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+NCPUS=${1}
+if [[ -z "${NCPUS}" ]]; then
+    echo "Missing number of CPUs to use" >&2
+    exit 1
+fi
+
+TARGETDIR=${2}
+if [[ -z "${TARGETDIR}" ]]; then
+    echo "Missing target directory argument" >&2
+    exit 1
+fi
+
+WORKDIR=/var/tmp/bitrot-detect
+mkdir -p ${WORKDIR}
+
+# Clean up previous run.
+rm -f ${WORKDIR}/tbs.md5-*
+
+find ${TARGETDIR}/ -maxdepth 1 -type d -printf '%P\n' | sort | while read ctrl; do
+    cat ${TARGETDIR}/${ctrl}/*.md5 2> /dev/null | while read hash file; do
+        echo "${hash} ${TARGETDIR}/${ctrl}/${file}"
+    done
+done > ${WORKDIR}/tbs.md5 2> ${WORKDIR}/tbs.err
+
+cat ${WORKDIR}/tbs.err; echo ""
+
+# We use the `--number=r/N` form so that all the workers below will
+# chew through the list together in alphabetical order, and reduces
+# the possibility of one workre getting all the biggest tar balls to
```

Comment:
`s/workre/worker`

And since you're sorting by filename, not by size, doesn't seem the order has any meaningful correlation to size: whether the splits or sequential or round-robin, the _size_ distribution is really random.

(Although I suppose that's not entirely true, since you'll be throwing the tarballs for "unnamed but well known users who typically generate enormous tarballs" across your workers... but if that's the point, maybe explaining that directly would make more sense of this.)

Reply:
We _could_ of course sort the tarballs by **size** and distribute them round-robin....

Reply:
Bad comment.  Reworded the comment to match the code and the intention of the code.